src/corelibs/U2View/src/ov_msa/view_rendering/SequenceAreaRenderer.cpp

```diff
@@ -78,36 +78,43 @@ bool SequenceAreaRenderer::drawContent(QPainter &painter, const U2Region &column
 #define SELECTION_STROKE_WIDTH 2
 
 void SequenceAreaRenderer::drawSelection(QPainter &painter) const {
-    QRect selectionRect = ui->getDrawHelper()->getSelectionScreenRect(seqAreaWgt->getEditor()->getSelection());
-    int viewWidth = ui->getSequenceArea()->width();
-    if (selectionRect.right() < 0 || selectionRect.left() > viewWidth) {
-        return;    // Selection is out of the screen.
-    }
-
-    // Check that frame has enough space to be drawn on both sides.
-    if (selectionRect.left() >= 0 && selectionRect.left() < SELECTION_STROKE_WIDTH) {
-        selectionRect.setLeft(SELECTION_STROKE_WIDTH);
-    }
-    if (selectionRect.right() <= viewWidth && selectionRect.right() + SELECTION_STROKE_WIDTH > viewWidth) {
-        selectionRect.setRight(viewWidth - SELECTION_STROKE_WIDTH);
-    }
+    const MaEditorSelection &selection = seqAreaWgt->getEditor()->getSelection();
+    CHECK(!selection.isEmpty(), );
 
+    painter.save();
     QPen pen(seqAreaWgt->selectionColor);
+    pen.setWidth(SELECTION_STROKE_WIDTH);
     if (seqAreaWgt->maMode == MaEditorSequenceArea::ViewMode) {
         pen.setStyle(Qt::DashLine);
     }
-    pen.setWidth(SELECTION_STROKE_WIDTH);
     painter.setPen(pen);
 
-    switch (seqAreaWgt->maMode) {
-        case MaEditorSequenceArea::ViewMode:
-        case MaEditorSequenceArea::ReplaceCharMode:
-            painter.drawRect(selectionRect);
-            break;
-        case MaEditorSequenceArea::InsertCharMode:
-            painter.drawLine(selectionRect.left(), selectionRect.top(), selectionRect.left(), selectionRect.bottom());
-            break;
+    const QList<QRect> selectionRects = selection.getRectList();
+    for (const QRect &selectionRect : qAsConst(selectionRects)) {
+        QRect screenRect = ui->getDrawHelper()->getScreenRect(selectionRect);
+        int viewWidth = ui->getSequenceArea()->width();
+        if (screenRect.right() < 0 || screenRect.left() > viewWidth) {
+            continue;    // Selection is out of the screen.
+        }
+
+        // Check that frame has enough space to be drawn on both sides.
+        if (screenRect.left() >= 0 && screenRect.left() < SELECTION_STROKE_WIDTH) {
```

Comment:
I don't like the way you text double comparasings (and comaprasings at all). You have both "<" and ">" signs, but, from my viewpoint, its better always use "<", because the abscissa increases from left to right and it's easier to imagine the mutual arrangement of variable's values if they increases from left to right.

I won't insist on this, but I will use my way

Reply:
I see,
0 < x && x < 10  is also right, but looks like a Yoda style and is not the way we speak in natural language.

I'm trying to write code that can be read in natural language in a straightforward way without any mental transformation involved.